src/api/java/com/minecolonies/api/util/ItemStackUtils.java

```diff
@@ -303,51 +303,34 @@ else if (Compatibility.isTinkersTool(stack, toolType))
             return Compatibility.getToolLevel(stack);
         }
 
-        if (ToolType.HOE.equals(toolType))
-        {
-            if (stack.getItem() instanceof HoeItem)
-            {
-                final HoeItem hoeItem = (HoeItem) stack.getItem();
-                return hoeItem.getTier().getLevel();
-            }
-        }
-        else if (ToolType.SWORD.equals(toolType))
-        {
-            if (stack.getItem() instanceof SwordItem)
-            {
-                final SwordItem SwordItem = (SwordItem) stack.getItem();
-                return SwordItem.getTier().getLevel();
-            }
-
-        }
-        else if (ToolType.HELMET.equals(toolType)
-                   || ToolType.BOOTS.equals(toolType)
-                   || ToolType.CHESTPLATE.equals(toolType)
-                   || ToolType.LEGGINGS.equals(toolType))
+        if (ToolType.HELMET.equals(toolType)
+                || ToolType.BOOTS.equals(toolType)
+                || ToolType.CHESTPLATE.equals(toolType)
+                || ToolType.LEGGINGS.equals(toolType))
         {
             if (stack.getItem() instanceof ArmorItem)
             {
                 final ArmorItem ArmorItem = (ArmorItem) stack.getItem();
                 return getArmorLevel(ArmorItem.getMaterial());
             }
         }
-        else if (stack.getItem() instanceof FishingRodItem)
+        else if (stack.getItem() instanceof TieredItem)  // most tools
```

Comment:
pattern matching
(btw instanceof variables can be finalized like normal ones)

Reply:
That was mostly on lines I wasn't changing, but sure.